sample_files/pre-commit-13.0/.pre-commit-config.yaml

```diff
@@ -2,67 +2,77 @@ exclude: "^setup/|/static/lib/|/static/src/lib/"
-    args: ["--rcfile=.pylintrc-mandatory"]
-    additional_dependencies: ["pylint-odoo==3.0.3"]
-- repo: https://github.com/asottile/pyupgrade
-  rev: v1.24.0
-  hooks:
-  - id: pyupgrade
-- repo: https://github.com/asottile/seed-isort-config
-  rev: v1.9.3
-  hooks:
-  - id: seed-isort-config
-- repo: https://github.com/pre-commit/mirrors-isort
-  rev: v4.3.21
-  hooks:
-  - id: isort
-    name: isort except __init__.py
-    exclude: /__init__\.py$
-- repo: https://github.com/pre-commit/mirrors-eslint
-  rev: v6.5.1
-  hooks:
-  - id: eslint
-    verbose: true
+  - repo: https://github.com/psf/black
+    rev: 19.3b0
+    hooks:
+      - id: black
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+      - id: trailing-whitespace
+        # exclude autogenerated files
+        exclude: /README\.rst$|\.pot?$
+      - id: end-of-file-fixer
+        # exclude autogenerated files
+        exclude: /README\.rst$|\.pot?$
+      - id: debug-statements
+      - id: flake8
+        name: flake8 except __init__.py
+        exclude: /__init__\.py$
+        additional_dependencies: ["flake8-bugbear==19.8.0"]
+      - id: flake8
+        name: flake8 only __init__.py
+        args: ["--extend-ignore=F401"] # ignore unused imports in __init__.py
+        files: /__init__\.py$
+        additional_dependencies: ["flake8-bugbear==19.8.0"]
+      - id: fix-encoding-pragma
+        args: ["--remove"]
+      - id: check-case-conflict
+      - id: check-docstring-first
+      - id: check-executables-have-shebangs
+      - id: check-merge-conflict
+      - id: check-symlinks
+      - id: check-xml
+      - id: mixed-line-ending
+        args: ["--fix=lf"]
+  - repo: https://github.com/pre-commit/mirrors-pylint
+    rev: v2.3.1
+    hooks:
+      - id: pylint
+        name: pylint with optional checks
+        args: ["--rcfile=.pylintrc", "--exit-zero"]
+        verbose: true
+        additional_dependencies: ["pylint-odoo==3.0.3"]
+      - id: pylint
+        name: pylint with mandatory checks
+        args: ["--rcfile=.pylintrc-mandatory"]
+        additional_dependencies: ["pylint-odoo==3.0.3"]
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v1.24.0
+    hooks:
+      - id: pyupgrade
+  - repo: https://github.com/asottile/seed-isort-config
+    rev: v1.9.3
+    hooks:
+      - id: seed-isort-config
+  - repo: https://github.com/pre-commit/mirrors-isort
+    rev: v4.3.21
+    hooks:
+      - id: isort
+        name: isort except __init__.py
+        exclude: /__init__\.py$
+  # TODO Add XML prettier plugin when it works fine, as explained in
+  # https://github.com/OCA/maintainer-quality-tools/pull/618
+  - repo: https://github.com/prettier/prettier
+    rev: "1.19.1"
+    hooks:
+      - id: prettier
+        exclude: /static/(lib|description)/
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    rev: v6.5.1
+    hooks:
+      - id: eslint
+        verbose: true
+        args: [--fix]
```

Comment:
Why this `--fix`? Is it not risky to have too tools attempting to reformat the same files?

Reply:
eslint has stylistic and programmatic checks, while prettier only has stylistic ones.

I disabled eslint's stylistic checks, so any fix done here will be only programmatic. If there's such conflict in the future, the fix will be to remove the stylistic check that eslint is using from `.eslintrc`, but I hope it doesn't happen.

I'm using https://github.com/prettier/eslint-config-prettier, which disables all rules that would conflict with prettier automatically (and that's why I removed some of them from `.eslintrc`). However let me take another look to make sure I disabled all.

Reply:
Nice, I removed some others that were remaining in our file, but found in https://github.com/prettier/eslint-config-prettier/blob/master/index.js.